mautic/mautic
# Add E2E tests for Segments

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description
**=== This is just a draft to show that I am working on tests for segments and to allow feedback. More tests will be added ===**
This PR adds e2e tests for segments. It also refactors some of the existing test to be more coherent and to generalise reusable functions.







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Verify that the all E2E-tests work by running `echo -e "APP_ENV=test\nAPP_DEBUG=1" > .env.local` and `ddev exec bin/codecept run acceptance`
